docs/source/developers/java/building.rst

```diff
@@ -107,33 +111,48 @@ Maven
         $ cd arrow/java
         $ export JAVA_HOME=<absolute path to your java home>
         $ java --version
-        $ mvn clean generate-resources -Pgenerate-cdata-dylib_so -N
+        $ mvn generate-resources -Pgenerate-libs-cdata-all-os -N
+        $ # MacOS / Linux
         $ ls -latr ../java-dist/lib
         |__ libarrow_cdata_jni.dylib
+        |__ libarrow_cdata_jni.so
+        $ # Windows
+        C:\ dir "../java-dist/bin"
+        |__ arrow_cdata_jni.dll
 
-- To build all JNI libraries except the JNI C Data Interface library:
+- To build all JNI libraries (MacOS / Linux) except the JNI C Data Interface library:
 
     .. code-block::
 
         $ cd arrow/java
         $ export JAVA_HOME=<absolute path to your java home>
         $ java --version
-        $ mvn clean generate-resources -Pgenerate-jni-dylib_so -N
+        $ mvn generate-resources -Pgenerate-libs-jni-macos-linux -N
         $ ls -latr java-dist/lib/*_{jni,java}.*
         |__ libarrow_dataset_jni.dylib
         |__ libarrow_orc_jni.dylib
         |__ libgandiva_jni.dylib
         |__ libplasma_java.dylib
 
+- To build all JNI libraries (Windows) except the JNI C Data Interface library:
+
+    .. code-block::
+
+        C:\ cd arrow/java
+        C:\ java --version
+        C:\ mvn generate-resources -Pgenerate-libs-jni-windows -N
+        C:\ dir "../java-dist/bin"
+        |__ arrow_dataset_jni.dll
+
 CMake
 ~~~~~
 
-- To build only the JNI C Data Interface library:
+- To build only the JNI C Data Interface library (MacOS / Linux):
 
     .. code-block::
 
         $ cd arrow
-        $ mkdir -p java-dist java-cdata
+        $ mkdir -p java-dist java-jni
```

Comment:
This seems like an odd name-change since cdata is much more descriptive

Reply:
Changed